File: library/src/app/model/UserBooks.ts
import { Book } from "./Book";
import { User } from "./User";

export class UserBooks{
    public id : number;
    public user : User;
    public book : Book;
    public issuedDate : Date;
    public returnDate : Date;
    
    constructor(id : number,user : User, book : Book,issueDate : Date,returnDate : Date){
        this.id = id;
        this.user = user;
        this.book = book;
        this.issuedDate = issueDate;
        this.returnDate = returnDate;
    }

}

File: backend/src/main/java/com/springboot/library/entity/UserBooks.java
package com.springboot.library.entity;

import java.sql.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name="user_books")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserBooks {

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;
	
	
	@ManyToOne
	@JoinColumn(name="user_id")
	private User user;
	
	@ManyToOne
	@JoinColumn(name="book_id")
	private Book book;
	
	private Date issuedDate;
	
	private Date returnDate;
	
	public UserBooks(User user, Book book) {
		this.user = user;
		this.book = book;
	}
	
}
